From de4632f47071ec417f2191ab6bebe73fa199c6b5 Mon Sep 17 00:00:00 2001
From: Lilibeth <lilibethocando@gmail.com>
Date: Wed, 1 May 2024 19:55:02 -0500
Subject: [PATCH] Updated client3.py

---
 client3.py | 13 +++++++------
 1 file changed, 7 insertions(+), 6 deletions(-)

diff --git a/client3.py b/client3.py
index 3fc09b7..39ad75e 100644
--- a/client3.py
+++ b/client3.py
@@ -31,18 +31,18 @@ N = 500
 
 def getDataPoint(quote):
     """ Produce all the needed values to generate a datapoint """
-    """ ------------- Update this function ------------- """
     stock = quote['stock']
     bid_price = float(quote['top_bid']['price'])
     ask_price = float(quote['top_ask']['price'])
-    price = bid_price
+    price = bid_price + ask_price / 2
     return stock, bid_price, ask_price, price
 
 
 def getRatio(price_a, price_b):
     """ Get ratio of price_a and price_b """
-    """ ------------- Update this function ------------- """
-    return 1
+    if price_b == 0:
+        price_b = 1
+    return price_a/price_b
 
 
 # Main
@@ -51,9 +51,10 @@ if __name__ == "__main__":
     for _ in iter(range(N)):
         quotes = json.loads(urllib.request.urlopen(QUERY.format(random.random())).read())
 
-        """ ----------- Update to get the ratio --------------- """
+        prices = {}
         for quote in quotes:
             stock, bid_price, ask_price, price = getDataPoint(quote)
+            prices[stock] = price
             print("Quoted %s at (bid:%s, ask:%s, price:%s)" % (stock, bid_price, ask_price, price))
 
-        print("Ratio %s" % getRatio(price, price))
+        print("Ratio %s" % getRatio(prices["ABC"], prices["DEF"]))
-- 
2.39.3 (Apple Git-146)

