From 532d8eb552221f807b29933e7c19f2304d2a37db Mon Sep 17 00:00:00 2001
From: Reyluvmachalatte <145343930+Reyluvmachalatte@users.noreply.github.com>
Date: Tue, 21 Nov 2023 19:46:44 +0700
Subject: [PATCH] Task 1 patch file

---
 client3.py | 11 ++++++++---
 1 file changed, 8 insertions(+), 3 deletions(-)

diff --git a/client3.py b/client3.py
index 3fc09b7..d535ba5 100644
--- a/client3.py
+++ b/client3.py
@@ -35,14 +35,19 @@ def getDataPoint(quote):
     stock = quote['stock']
     bid_price = float(quote['top_bid']['price'])
     ask_price = float(quote['top_ask']['price'])
-    price = bid_price
+    price =  (bid_price+ask_price) / 2
     return stock, bid_price, ask_price, price
 
 
 def getRatio(price_a, price_b):
     """ Get ratio of price_a and price_b """
     """ ------------- Update this function ------------- """
-    return 1
+    """ Also create some unit tests for this function in client_testpy"""
+    if (priice_b == 0):
+        # when price_b is 0 avoid throwing ZeroDivisionError
+        return
+    return price_a/price_b
+    
 
 
 # Main
@@ -56,4 +61,4 @@ if __name__ == "__main__":
             stock, bid_price, ask_price, price = getDataPoint(quote)
             print("Quoted %s at (bid:%s, ask:%s, price:%s)" % (stock, bid_price, ask_price, price))
 
-        print("Ratio %s" % getRatio(price, price))
+        print("Ratio %s" % getRatio(price["ABC"], price["DEF"]))                                                  
-- 
2.39.2 (Apple Git-143)

