From 680097fb556278c095becc9c952730316d53a4bc Mon Sep 17 00:00:00 2001
From: Dawde Seifu <seifudawde@gmail.com>
Date: Wed, 11 Oct 2023 20:14:17 +0300
Subject: [PATCH] update the client3

---
 client3.py | 11 +++++++----
 1 file changed, 7 insertions(+), 4 deletions(-)

diff --git a/client3.py b/client3.py
index 3fc09b7..9500753 100644
--- a/client3.py
+++ b/client3.py
@@ -35,15 +35,17 @@ def getDataPoint(quote):
     stock = quote['stock']
     bid_price = float(quote['top_bid']['price'])
     ask_price = float(quote['top_ask']['price'])
-    price = bid_price
+    price = (bid_price + ask_price) / 2
     return stock, bid_price, ask_price, price
 
 
 def getRatio(price_a, price_b):
     """ Get ratio of price_a and price_b """
     """ ------------- Update this function ------------- """
-    return 1
-
+    if (price_b==0):
+        return
+    ratio = price_a / price_b
+    return ratio
 
 # Main
 if __name__ == "__main__":
@@ -52,8 +54,9 @@ if __name__ == "__main__":
         quotes = json.loads(urllib.request.urlopen(QUERY.format(random.random())).read())
 
         """ ----------- Update to get the ratio --------------- """
+        prices = {}
         for quote in quotes:
             stock, bid_price, ask_price, price = getDataPoint(quote)
             print("Quoted %s at (bid:%s, ask:%s, price:%s)" % (stock, bid_price, ask_price, price))
 
-        print("Ratio %s" % getRatio(price, price))
+        print("Ratio %s" % getRatio(prices["ABC"], prices['DEF']))
-- 
2.35.1.windows.2

